src/ts/contentScripts/live-preview/index.tsx

```diff
@@ -0,0 +1,169 @@
+    }
+})
+
+browser.runtime.onMessage.addListener(async message => {
+    if (message?.featureId === 'live_preview' && message?.type === 'toggle') {
+        if (message.value) {
+            enableLivePreview()
+        } else {
+            removeLivePreview()
+        }
+    }
+})
+
+const createPreviewIframe = () => {
+    const iframe = document.createElement('iframe')
+
+    const url = Navigation.getPageUrl('search')
+    const isAdded = (pageUrl: string) => !!document.querySelector(`[src="${pageUrl}"]`)
+    if (isAdded(url)) {
+        return
+    }
+    iframe.src = url
+    iframe.style.position = 'absolute'
+    iframe.style.left = '0'
+    iframe.style.top = '0'
+    iframe.style.opacity = '0'
+    iframe.style.pointerEvents = 'none'
+
+    iframe.style.height = '0'
+    iframe.style.width = '0'
+    iframe.style.border = '0'
+    iframe.style.boxShadow = '0 0 4px 5px rgba(0, 0, 0, 0.2)'
+    iframe.style.borderRadius = '4px'
+    iframe.id = 'roam-toolkit-preview-iframe'
+
+    const styleNode = document.createElement('style')
+    styleNode.innerHTML = `
+        .roam-topbar {
+            display: none !important;
+        }
+        .roam-body-main {
+            top: 0px !important;
+        }
+        #buffer {
+            display: none !important;
+        }
+        iframe {
+            display: none !important;
+        }
+    `
+    iframe.onload = (event: Event) => {
+        ;(event.target as HTMLIFrameElement).contentDocument?.body.appendChild(styleNode)
+    }
+    document.body.appendChild(iframe)
+    const htmlElement = document.querySelector('html')
+    if (htmlElement) {
+        // to reset scroll after adding iframe
+        htmlElement.scrollTop = 0
+    }
+    return iframe
+}
+const enableLivePreview = () => {
+    let hoveredElement: HTMLElement | null
+    let popupTimeout: ReturnType<typeof setTimeout> | null
+    let popper: Instance | null = null
+    const previewIframe = createPreviewIframe()
+    document.addEventListener('mouseover', (e: Event) => {
+        const target = e.target as HTMLElement
+        const isPageRef = target.classList.contains('rm-page-ref')
+        const isPageRefTag = target.classList.contains('rm-page-ref-tag')
+        // remove '#' for page tags
+        const text = isPageRefTag ? target.innerText.slice(1) : target.innerText
+        if (isPageRef) {
+            hoveredElement = target
+            const url = Navigation.getPageUrlByName(text)
+            const isAdded = (pageUrl: string) => !!document.querySelector(`[src="${pageUrl}"]`)
+            const isVisible = (pageUrl: string) =>
+                (document.querySelector(`[src="${pageUrl}"]`) as HTMLElement)?.style.opacity === '1'
+            if ((!isAdded(url) || !isVisible(url)) && previewIframe) {
+                previewIframe.src = url
+                previewIframe.style.height = '500px'
+                previewIframe.style.width = '500px'
+                previewIframe.style.pointerEvents = 'none'
```

Comment:
whatever this is doing should be extracted into the function that has name that explain what it's doing